File: jbake-core/src/main/java/org/jbake/template/model/AllTagsExtractor.java
package org.jbake.template.model;

import org.jbake.app.ContentStore;
import org.jbake.model.TemplateModel;

import java.util.Set;

public class AllTagsExtractor implements ModelExtractor<Set<String>> {

    @Override
    public Set<String> get(ContentStore db, TemplateModel model, String key) {
        return db.getAllTags();
    }

}

File: jbake-core/src/main/java/org/jbake/template/model/AllTagsExtractor.kt
package org.jbake.template.model

import org.jbake.template.TypedModelExtractor

class AllTagsExtractor : TypedModelExtractor<MutableSet<String>> {

    override fun extract(context: RenderContext, key: String): MutableSet<String> {
        return context.db.allTags
    }
}
